ADAL/tests/integration/ios/ADBrokerIntegrationTests.m

```diff
@@ -193,6 +206,176 @@ - (void)testBroker_whenSimpleAcquireToken_shouldSucceed
+                                                        @"error_description" : @"AADSTS53005: Application needs to enforce intune protection policies",
+                                                        @"error" : @"unauthorized_client",
+                                                        @"suberror" : @"protection_policies_required",
+                                                        }];
+        NSDictionary *intune_token_response = @{
+                                              @"authority" : authority,
+                                              @"resource" : TEST_RESOURCE,
+                                              @"client_id" : TEST_CLIENT_ID,
+                                              @"id_token" : [[self adCreateUserInformation:TEST_USER_ID] rawIdToken],
+                                              @"access_token" : @"i-am-a-access-token",
+                                              @"refresh_token" : @"i-am-a-refresh-token",
+                                              @"foci" : @"1",
+                                              @"expires_in" : @"3600"};
+        NSDictionary* encrypted_token = [ADBrokerIntegrationTests createV2BrokerResponseDicitonary:intune_token_response];
+
+        [responseParams setValue:encrypted_token[BROKER_RESPONSE_KEY] forKey:BROKER_INTUNE_RESPONSE_KEY];
+        [responseParams setValue:encrypted_token[BROKER_HASH_KEY] forKey:BROKER_INTUNE_HASH_KEY];
+        [responseParams setValue:encrypted_token[@"msg_protocol_ver"] forKey:@"msg_protocol_ver"];
+
+        [ADAuthenticationContext handleBrokerResponse:[ADBrokerIntegrationTests createV2BrokerErrorResponse:responseParams redirectUri:redirectUri]];
+        return YES;
+    }];
+
+    NSArray *metadata = @[ @{ @"preferred_network" : @"login.microsoftonline.com",
+                              @"preferred_cache" : @"login.windows.net",
+                              @"aliases" : @[ @"login.windows.net", @"login.microsoftonline.com"] } ];
+    ADTestURLResponse *validationResponse =
+    [ADTestAuthorityValidationResponse validAuthority:authority
+                                          trustedHost:@"login.windows.net"
+                                         withMetadata:metadata];
+    [ADTestURLSession addResponses:@[validationResponse]];
+
+    ADAuthenticationContext *context = [self getBrokerTestContext:authority];
+    XCTestExpectation *expectation = [self expectationWithDescription:@"acquire token callback"];
+    [context acquireTokenWithResource:TEST_RESOURCE
+                             clientId:TEST_CLIENT_ID
+                          redirectUri:[NSURL URLWithString:redirectUri]
+                      completionBlock:^(ADAuthenticationResult *result)
+     {
+         XCTAssertNotNil(result);
+         XCTAssertEqual(result.status, AD_FAILED);
+         XCTAssertEqual(result.error.code, AD_ERROR_SERVER_PROTECTION_POLICY_REQUIRED);
```

Comment:
It is better to verify what is in the userInfo here, to make sure the error object contains user id, suberror, etc.